rcmd/helpers_test.go

```diff
@@ -0,0 +1,62 @@
+package rcmd
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBinaryName(t *testing.T) {
+	tests := []struct {
+		os       string
+		platform string
+		expected string
+	}{
+		{
+			os:       "linux",
+			platform: "x86_64-pc-linux-gnu",
+			expected: "pkg_version_R_x86_64-pc-linux-gnu.tar.gz",
+		},
+		{
+			os:       "darwin",
+			expected: "pkg_version.tgz",
+		},
+		{
+			os:       "windows",
+			expected: "pkg_version.zip",
+		},
+	}
+	for _, tt := range tests {
+		name := binaryNameOs(tt.os, "pkg", "version", tt.platform)
+		assert.Equal(t, tt.expected, name, fmt.Sprintf("Not equal: %s", tt.os))
+	}
+}
+
+func TestBinaryExt(t *testing.T) {
+	tests := []struct {
+		os       string
+		path     string
+		expected string
+	}{
+		{
+			os:       "linux",
+			path:     "/var/tmp/gz",
```

Comment:
background: 

starting path will be something like: `path/to/<package>_<version>.tar.gz`

but when R CMD BUILD --binary is run, it will spit out a binary that is named specific to the platform/OS.

So we need to convert the tarball path into what we'd want to look for binary-wise.

```
tar.gz --> .tgz for mac
tar.gz --> .zip for windows
```